### ISSUE TITLE
--jobs=0 no longer does any work

### ISSUE THREAD

**Issue by drewnoakes** [git-for-windows/git#4302]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

(Note that I did report this on the git mailing list [here](https://lore.kernel.org/git/PSAP153MB03910458707331B64FA7460DCAA19@PSAP153MB0391.APCP153.PROD.OUTLOOK.COM/T/#u), but no response was received.)

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.39.2.windows.1
cpu: x86_64
built from commit: a82fa99b36ddfd643e61ed45e52abe314687df67
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.22621.1265]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
Editor Option: VIM
Custom Editor Path:
Default Branch Option: main
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: FFOnly
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

Not that I know of.

### Details

 - Which terminal/shell are you running Git from?

Git Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git fetch --jobs=0
```
 - What did you expect to occur after running these commands?

Fetches from my multiple remotes to occur concurrently.

 - What actually happened instead?

Nothing was fetched.

Yesterday, nothing was logged. Today I see an error message (no idea what changed):

> BUG: run-command.c:1521: you must provide a non-zero number of processes!

Note also:

- The git-config documentation says of zero jobs in "fetch.parallel" that "A value of 0 will give some reasonable default. If unset, it defaults to 1.".
- This started happening after I updated my version of Git for Windows earlier today.
- Excluding the --jobs argument or passing a non-zero value works as expected.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Doesn't seem related to a specific repo.

**Comment by rimrul** [comment]
For completeness: This has also been reported [on the mailing list](https://lore.kernel.org/git/PSAP153MB03910458707331B64FA7460DCAA19@PSAP153MB0391.APCP153.PROD.OUTLOOK.COM/T/#u).


> run-command.c:1521

Ok, that's a starting point where to look.
And bingo. A [blame on run-command.c:L1521](https://github.com/git-for-windows/git/blame/49a8ec9dac3cec6602f05fed1b3f80a549c8c05c/run-command.c#L1521) tells us it was last changed in [51243f9](https://github.com/git/git/commit/51243f9f0f6a932ea579fd6f8014b348f8c2a523).

Looks like Ævar added the `online_cpus()` fallback int `fetch.c` for `fetch.parallel`, but missed `--jobs=0`. I'll try to write a fix in the next few days.

**Comment by dscho** [comment]
@rimrul since you seem to be almost there anyway, I added you as assignee, okay?

**Comment by rimrul** [comment]
> since you seem to be almost there anyway, I added you as assignee, okay?

yes. I've submitted the patch [to the mailing list](https://lore.kernel.org/git/pull.1483.git.1676928805555.gitgitgadget@gmail.com/).

**Comment by rimrul** [comment]
The fix has not made it in time for Git `2.40.0-rc0`, but will be part of `2.40.0-rc1`.
